frontend/middleware/driver/lib/api-router.ts

```diff
@@ -0,0 +1,218 @@
+        const { id, password } = req.body;
+        if (!id || !password || typeof id !== 'string' || typeof password !== 'string') {
+            return res.status(400).json({ ok: false, message: "IDとパスワードが必要です" });
+        }
+
+        try {
+            const userExists = await DevUserManager.hasUser();
+            if (userExists) {
+                return res.status(409).json({ ok: false, message: "ユーザーは既に登録されています" });
+            }
+
+            await DevUserManager.createUser(id, password);
+
+            // サインアップ後、自動的にログインセッションを開始
+            await new Promise<void>((resolve, reject) => {
+                req.session.regenerate((err) => {
+                    if (err) return reject(err);
+                    req.session.userId = id;
+                    req.session.loginAt = new Date().toISOString();
+                    req.session.save((saveErr) => saveErr ? reject(saveErr) : resolve());
+                });
+            });
+
+            return res.status(201).json({ ok: true, message: "ユーザー登録とログインが完了しました", userId: id });
+        } catch (error) {
+            console.error("Signup error:", error);
+            return res.status(500).json({ ok: false, message: "サーバー内部エラーが発生しました" });
+        }
+    };
+
+    private loginHandler: express.RequestHandler = async (req, res) => {
+        const { id, password } = req.body;
+        if (!id || !password) {
+            return res.status(400).json({ ok: false, message: "IDとパスワードが必要です" });
+        }
+
+        try {
+            const authenticatedUserId = await DevUserManager.authenticate(id, password);
+            if (!authenticatedUserId) {
+                return res.status(401).json({ ok: false, message: "IDまたはパスワードが正しくありません" });
+            }
+
+            await new Promise<void>((resolve, reject) => {
+                req.session.regenerate((err) => {
+                    if (err) return reject(err);
+                    req.session.userId = authenticatedUserId;
+                    req.session.loginAt = new Date().toISOString();
+                    req.session.save((saveErr) => saveErr ? reject(saveErr) : resolve());
+                });
+            });
+            console.log(`User logged in: ${authenticatedUserId} at ${req.session.loginAt}`);
+
+            return res.status(200).json({ ok: true, message: "ログインに成功しました", userId: authenticatedUserId });
+        } catch (error) {
+            console.error("Login error:", error);
+            return res.status(500).json({ ok: false, message: "サーバー内部エラーが発生しました" });
+        }
+    };
+
+    private authHandler: express.RequestHandler = (req, res) => {
+        if (req.session?.userId) {
+            return res.status(200).json({
+                ok: true,
+                userId: req.session.userId,
+                loginAt: req.session.loginAt,
+                message: "認証済みです"
+            });
+        }
+        // ユーザーがまだ登録されていない状態も考慮
+        DevUserManager.hasUser().then(hasUser => {
```

Comment:
Should use "await" method to connect upper "async" method.

Also should use try/ catch for error handling.

Like this : 

try {
            const hasUser = await DevUserManager.hasUser();
            console.log("Auth check: hasUser =", hasUser);
            if (!hasUser) {
                return res.status(200).json({ ok: false, reason: "no_user_registered", message: "ユーザーが登録されていません" });
            }
            return res.status(401).json({ ok: false, reason: "invalid_session", message: "セッションが無効です" });
        } catch (error) {
            console.error("Auth check error:", error);
            return res.status(500).json({ ok: false, message: "サーバー内部エラーが発生しました" });

Reply:
```suggestion
    private static async getUser(): Promise<User | null> {
        try {
            const usersData = await fs.readFile(USERS_FILE, 'utf-8');
            const user = JSON.parse(usersData);
            console.log("Read user data:", user);
            if (!user.id || !user.passwordHash) {
                return null;
            } else {
                return user as User | null;
            }

        } catch (error) {
            console.error("Failed to read user file:", error);
            return null;
        }
    }
```
こんな感じで
呼び出した奥の関数でエラーハンドリングしてるから
大した問題は起きないと思う